src/protocol/boolean/bit_decomposition.rs

```diff
@@ -0,0 +1,209 @@
+use super::bitwise_lt::BitwiseLessThan;
+use super::bitwise_sum::BitwiseSum;
+use crate::error::Error;
+use crate::ff::{Field, Int};
+use crate::protocol::boolean::local_secret_shared_bits;
+use crate::protocol::boolean::solved_bits::SolvedBits;
+use crate::protocol::reveal::Reveal;
+use crate::protocol::{context::ProtocolContext, RecordId};
+use crate::secret_sharing::Replicated;
+
+/// This is an implementation of "3. Bit-Decomposition" from I. Damgård et al..
+///
+/// It takes an input `[a] ∈ F_p` and outputs its bitwise additive share
+/// `[a]_B = ([a]_0,...,[a]_l-1)` where `[a]_i ∈ F_p`.
+///
+/// 3. Bit-Decomposition
+/// "Unconditionally Secure Constant-Rounds Multi-party Computation for Equality, Comparison, Bits, and Exponentiation"
+/// I. Damgård et al.
+pub struct BitDecomposition {}
+
+impl BitDecomposition {
+    #[allow(dead_code)]
+    pub async fn execute<F: Field>(
+        ctx: ProtocolContext<'_, Replicated<F>, F>,
+        record_id: RecordId,
+        a_p: &Replicated<F>,
+    ) -> Result<Vec<Replicated<F>>, Error> {
+        // step 1 in the paper is just describing the input, `[a]_p` where `a ∈ F_p`
+
+        // Step 2. Generate random bitwise shares
+        let r = SolvedBits::execute(ctx.narrow(&Step::GenerateRandomBits), record_id).await?;
+        // Would like to do something like below. Where should I add the `RandomBitsGenerator`?
+        // let r = ctx.random_bits_generator.take_one().await?;
```

Comment:
Need to place the buffer instance somewhere. Suggestions are much appreciated :)

Reply:
The commented out lines look good - definitely better than `r.expect()` which will just fail if it aborted.